### ISSUE TITLE
Consider removing `rebuildShares`

### ISSUE THREAD

**Issue by rootulp** [celestiaorg/rsmt2d#199]
do we need this function at all if its just decoding?

_Originally posted by @evan-forbes in https://github.com/celestiaorg/rsmt2d/pull/197#discussion_r1245341961_

**Comment by rootulp** [comment]
blocked on https://github.com/celestiaorg/rsmt2d/issues/200

**Comment by rootulp** [comment]
Until the error handling logic isn't janky, I think it actually makes sense to retain `rebuildShares` as a stand-alone function so closing as won't fix.
